src/nccl_ofi_net.c

```diff
@@ -1197,6 +1197,17 @@ static ncclResult_t update_nccl_topology()
 	return ret;
 }
 
+static void ofi_read_num_gpus(int *count){
+	char *fi_efa_num_gpus = getenv("FI_EFA_NUM_GPUS");
+	if(fi_efa_num_gpus) {
+		*count = atoi(fi_efa_num_gpus);
```

Comment:
Use `strtol` instead to prevent user entering any garbage value?

Reply:
changed it to `strtoi` instead